dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/ContextFilter.java

```diff
@@ -80,14 +80,14 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         } finally {
             // IMPORTANT! For async scenario, we must remove context from current thread, so we always create a new RpcContext for the next invoke for the same thread.

```

Comment:
Didn't you see this comment?

We must remove the context at the end of the invoke. So I think your pr will cause new problems.

Reply:
How about use a local variable to store the async identifier, and in the finally block, we can check if it is a async invoke, the servercontext will be removed. If not, we donnot remove it. Doesn't this impact the asyn invoke ?